### ISSUE TITLE
Proton & Steam Game Compatibility/Support

### ISSUE THREAD

**Issue by BlueGoliath** [ValveSoftware/Proton#3123]
(Note: tried asking via Steam support but there is no way to ask general questions, so asking here.)

Hi Valve,

Recently [a thread](https://www.reddit.com/r/linux/comments/dcu7pb/bungie_will_permanently_ban_anyone_who_plays/?st=k1c6mn4r&sh=08d36d97) was started on the /r/linux subreddit which contained many slanderous statements towards a game developer/publisher because users were using a modified Proton version to bypass in-game anti cheat. This isn't even the first time, actually: many highly upvoted threads and comments have been made to that subreddit and elsewhere regarding this specific game banning Proton users for bypassing anti-cheat. The moderators of that subreddit have even removed threads/comments pointing out that the user of Proton used modified Wine patches.

Developers are getting support tickets from users who are attempting to play on unsupported platforms and then are using those same support tickets as weapons to unjustly defame these developers/pubishers when Linux was never a support platform to begin with. This is highly unfair and will likely result in negative articles and bad PR when the developer/publisher did nothing wrong.

So, my questions fall into two general categories:

A. What rights do users of Proton have? Can you buy a game on Linux with the intention of attempting to play it using Proton and still get a refund? What if a developer introduced a Proton breaking change to their game and you are outside the standard refund period?

B. What rights and protections do developers have with Proton? Do they have the right to ban a customer because they are using Proton? What if they are working around anti-cheat like in this case? Are customers still allowed to get a refund in such cases?

C. Does Valve know about this harassment and slander against developers who ban Proton users for working around anti-cheat? Does Valve think that it is OK?

Thanks for any clarification.

**Comment by fls2018** [comment]
There was no modified proton version AFAIK, the original efforts to get this working are on the wine [bugzilla](https://bugs.winehq.org/show_bug.cgi?id=44061) known as the kimmo patch and that was what led to bans at the time. 

I don't think it's slanderous to warn people they might get banned if they attempt to run it, fact is if wine/proton is somehow accidentally or intentionally patched in future to allow the game to run people who don't know any better might end up with bans because there are no prior obvious warnings to say it's not allowed and it hasn't been OK'ed by the developer.. whom are pretty [clear](https://www.bungie.net/en/Forums/Post/253781012?sort=0&page=0&path=1) on what they ban for.

**Comment by rdunski** [comment]
Might want to move this convo to #3112. Don't want to muddy Valve's issue pages. But in Bungie's defense, it would take some non-insignificant amount of work in order to provide compatibility for Proton, and that might mean rewriting their anti-cheat software to allow Linux users to play or even find a solution for themselves. They're not being aggressive and actively seeking to ban all linux users, their anti-cheat is just automating the process. Why they won't un-ban those that were proven to not be cheating is another discussion.

**Comment by BlueGoliath** [comment]
>There was no modified proton version AFAIK, the original efforts to get this working are on the wine bugzilla known as the kimmo patch and that was what led to bans at the time.

Ah, thought it was DLL modifications or something. For some games it could still be the case that it is that simple.

Regardless of the context, people will lash out whenever a game developer/publisher breaks a game that worked on Proton before.

>I don't think it's slanderous to warn people they might get banned if they attempt to run it, fact is if wine/proton is somehow accidentally or intentionally patched in future to allow the game to run people who don't know any better might end up with bans because there are no prior obvious warnings to say it's not allowed and it hasn't been OK'ed by the developer.. whom are pretty clear on what they ban for.

You'l need to scroll past the OP to find the comments:

[1](https://www.reddit.com/r/linux/comments/dcu7pb/bungie_will_permanently_ban_anyone_who_plays/f2cynev/?st=k1cdghah&sh=a8fd9dd7)

[2](https://www.reddit.com/r/linux/comments/dcu7pb/bungie_will_permanently_ban_anyone_who_plays/f2cdzqk/?st=k1cdia18&sh=1ee3efa4)

[3](https://www.reddit.com/r/linux/comments/dcu7pb/bungie_will_permanently_ban_anyone_who_plays/f2bitzc/?st=k1cdmwe2&sh=afa37295)

those are only 3 that I found by lazily searching for "Bungie". God knows how many more.

The thing is, people don't care about the provided reason. They just blame the developers/publishers anyway. All it takes is some journalist not reading that and next thing you know you'l see an article titled "Destiny 2 developer bans Linux users".

**Comment by BlueGoliath** [comment]
>Might want to move this convo to #3112. Don't want to muddy Valve's issue pages.

This is an issue that could easily be applied to many games. 

For example, is it fair for a game developer to have to refund a game because it doesn't work on an unsupported platform and the customer knew that it probably wouldn't beforehand?

Which is why I'm asking Valve for some clarifications.

**Comment by rdunski** [comment]
Well technically the developers/company who made the game wouldn't be held accountable if they don't officially support Linux. Most Linux gamers should realize that not everything is going to work on the OS because we're in the minority and assess the risks accordingly. Granted, I'm not a lawyer, and I'm not associated with Valve, so this is just my opinion.

**Comment by BlueGoliath** [comment]
You already have YouTubers talking about Bungie banning Destiny 2 players with sensationalist YouTube video titles like [this](https://www.youtube.com/watch?v=xtl9yjaomQA).

**Comment by fls2018** [comment]
> You already have YouTubers talking about Bungie banning Destiny 2 players with sensationalist YouTube video titles like [this](https://www.youtube.com/watch?v=xtl9yjaomQA).

I don't know why you're calling it sensationalist, everything described in TLG's video is accurate i.e. if you press play on the game with Proton normally it won't launch but if you patch or modify it to run then you will end up with a ban. He also rubbished the redditor Microsoft-Bungie conspiracy theories in that video. 

The majority aren't mad because they think this is a ruse to prevent them gaming on Linux, they are annoyed at the fact by using a tool (Proton) that's embedded into the games new launcher (Steam) holds the possibility that they might lose access to the game entirely (rather than just get kicked like with EAC/Battleye).

As stated in my last post it holds the possibility something might change with Proton in future that might allow the game to launch but users might end up banned as a result.

Not saying Bungie are bad because they don't want to support Proton as that's their right not to support specific platforms, however there is no denfinitate 'block' in place currently preventing users from launching the game with Proton (other than the crash) and that might lead to all sorts of banning related issues.

**Comment by ghost** [comment]
Its not really up to Valve to put up a warning that bans may result (Bungie could anyway on the store page) nor is it for them to block Proton from being used. They could do all or any of these things, but at least to me, reasonably its not for them. The game is not their product, they do not issue the bans, etc. Bungie does [in this case].

I would say that if Bungie bans for using Proton then talk to Bungie that you didn't know it was cheating and ask them to talk to Valve for a refund or an un-ban. I feel the best other alternative is to have Valve give developers access to a way to disable Proton in special cases. It might even be possible for Bungie to check if Proton is being used (I don't know if Proton would be labelled from game code in some way) and then block it pre-menu.

Sure, Proton is part of Valve's service and it allows even broken games to be played should they work, etc. Valve could restrict it to only whitelisted games and put up a warning that way for game's that aren't whitelisted so that it can still be done. Perhaps there's some legal thing there that even broken games can be played but Wine has always been this way and I don't know of any legal issues there.

**Comment by Csardelacal** [comment]
I really have a gripe with you considering anything that was going on on that reddit as "harassment". Three guys dropped generalized statements about them and a bunch of f-bombs... And you come here brigading for having Valve make changes to their software? Wine / Proton is intended to provide compatibility with Windows and enable access to people who for some reason do not wish to have a Windows machine.

> You already have YouTubers talking about Bungie banning Destiny 2 players with sensationalist YouTube video titles like [this](https://www.youtube.com/watch?v=xtl9yjaomQA).

The title of the video is technically correct. The reason the bans were dished out was because the machines were running Wine, and not genuine Windows.

About your point about refunds, if the game doesn't provide the service it advertised to me, I expect it to be refundable. Period. I wouldn't expect it to be refunded past the 2 hour mark, but if I buy the game and it is either borked / unplayable, I would not expect to be required to keep it.

I would applaud Valve if they just closed this issue and deleted it. It's pointless, and doesn't add anything.

**Comment by SimplyCorbett** [comment]
@BlueGoliath 

You risk getting banned using Wine with any online game. This is also true for VFIO/KVM users.

IMO the issue has nothing to do with steam and more to do with how people are swapping from windows to easy to install distros such as Manjaro/PopOS and expect it to behave just like windows. 

Steam makes it easy to use wine/proton for new users. But that doesn't mean you shouldn't be doing your own research and due diligence into what wine/proton is, how it works and if you risk getting banned for using it.

Steam is also very clear about it being a compatibility tool on every game you launch using Wine/Proton.

Steam also only whitelists a number of games running on proton. To run other games you have to bypass the whitelist. Steam shouldn't have to spell out to you that unsupported/untested games mean they're unsupported/untested.

tl;dr I don't understand why you're making a fuss about this. Just because proton is now included in steam doesn't mean every game/developer is going to magically support it.

**Comment by Lyle-Tafoya** [comment]
"because users were using a modified Proton version to bypass in-game anti cheat"

That is not what I got out of that thread at all. Their response is pretty clear cut: "Destiny 2 on PC only supports Windows. Linux via Proton is not supported and is considered a modified operating system, as is Wine and other emulation services". That doesn't sound like the status of their anticheat has anything to do with it. I can only assume that you didn't read their full response before opening this issue. They ban people for using modified operating systems. They consider Proton to be a modified operating system. End of story.

**Comment by xsgb** [comment]
I think If the user is warned about the anti-cheat system not supporting wine, then he's responsible of being banned automatically and is not in the right position to send negative critics on the developers. 

However, technically, I think developers should do a simple check to warn the user about the presence of AC running on unsupported platform before doing anything. That would be welcome and opening long-term way to let the community and the developers to work together on the Anti-Cheat problem.

**Comment by OvermindDL1** [comment]
> I think If the user is warned about the anti-cheat system not supporting wine, then he's responsible of being banned automatically and is not in the right position to send negative critics on the developers.

I would not agree, being warned about a compat issue (which may or may not actually exist if the programmers are actually good at their job) is not the same as being called a cheater when you are not, which is how people being banned will take that.

**Comment by xsgb** [comment]
I have no technical background on how an AC could be implemented, but I see many way it won't be bullet-proof due to the differences in memory management and validating underlying API's between platforms. I think it's not easy.

However yes, being warned in the first time before actually running the AC during a party would be great and welcome.

**Comment by OvermindDL1** [comment]
> However yes, being warned in the first time before actually running the AC during a party would be great and welcome.

That would be on the game authors to implement however, whether internally in code or via a steam flag that they'd have to enable (which I *know* would get abused by others).  I don't trust Bethesda, EA, Activision, Epic, Etc... to do anything like that due to their displayed incompetence.
